x-pack/plugin/enrich/src/main/java/org/elasticsearch/xpack/enrich/EnrichStore.java

```diff
@@ -37,9 +63,8 @@ private EnrichStore() {}
      * @param policy    The policy to store
      * @param handler   The handler that gets invoked if policy has been stored or a failure has occurred.
      */
-    public static void putPolicy(String name, EnrichPolicy policy, ClusterService clusterService, Consumer<Exception> handler) {
-        assert clusterService.localNode().isMasterNode();
-
+    public static void putPolicy(String name, EnrichPolicy policy, ClusterService clusterService, Client client,
```

Comment:
Maybe replace the `ClusterService` parameter with `ClusterState` here and in other methods, since we only seem to invoke `ClusterService#state()` method now.